[2024-07-24T16:23:18.375+0900] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-24T16:23:18.382+0900] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-24T16:23:18.385+0900] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-24T16:23:18.385+0900] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-24T16:23:18.389+0900] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): to.parquet> on 2024-07-12 00:00:00+00:00
[2024-07-24T16:23:18.391+0900] {standard_task_runner.py:64} INFO - Started process 99063 to run task
[2024-07-24T16:23:18.394+0900] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'make_parquet', 'to.parquet', 'scheduled__2024-07-12T00:00:00+00:00', '--job-id', '420', '--raw', '--subdir', 'DAGS_FOLDER/make_parquet.py', '--cfg-path', '/var/folders/qv/0knwm5h50ml0skpppwpbks_r0000gp/T/tmpm3ppv464']
[2024-07-24T16:23:18.395+0900] {standard_task_runner.py:91} INFO - Job 420: Subtask to.parquet
[2024-07-24T16:23:18.409+0900] {task_command.py:426} INFO - Running <TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [running]> on host jeongui-MacBookPro.local
[2024-07-24T16:23:18.426+0900] {abstractoperator.py:725} ERROR - Exception rendering Jinja template for task 'to.parquet', field 'bash_command'. Template: '\n        echo "to.parquet"\n\n        READ_PATH=~/data/csv/{{ds_nodash}}/csv.csv\n        SAVE_PATH={{var.value.PARQUET_DONE_PATH}}/{{ds_nodash}}/parquet.parquet\n\n        mkdir -p $SAVE_PATH\n\n        python {{var.value.OP_HOME}}/csv2parquet.py $READ_PATH $SAVE_PATH\n        '
Traceback (most recent call last):
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 717, in _do_render_template_fields
    rendered_content = self.render_template(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/template/templater.py", line 169, in render_template
    return self._render(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 675, in _render
    return super()._render(template, context, dag=dag)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/template/templater.py", line 126, in _render
    return render_template_to_string(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
           ^^^^^^^^^^^^^^
  File "<template>", line 16, in root
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/jinja2/sandbox.py", line 327, in getattr
    value = getattr(obj, attribute)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/context.py", line 110, in __getattr__
    self.var = Variable.get(key, deserialize_json=self._deserialize_json)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/variable.py", line 143, in get
    raise KeyError(f"Variable {key} does not exist")
KeyError: 'Variable PARQUET_DONE_PATH does not exist'
[2024-07-24T16:23:18.428+0900] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2479, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2633, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context, jinja_env=jinja_env)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3094, in render_templates
    original_task.render_template_fields(context, jinja_env)
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 1337, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 717, in _do_render_template_fields
    rendered_content = self.render_template(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/template/templater.py", line 169, in render_template
    return self._render(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 675, in _render
    return super()._render(template, context, dag=dag)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/template/templater.py", line 126, in _render
    return render_template_to_string(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
           ^^^^^^^^^^^^^^
  File "<template>", line 16, in root
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/jinja2/sandbox.py", line 327, in getattr
    value = getattr(obj, attribute)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/utils/context.py", line 110, in __getattr__
    self.var = Variable.get(key, deserialize_json=self._deserialize_json)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/DE32/.pyenv/versions/3.11.9/envs/note/lib/python3.11/site-packages/airflow/models/variable.py", line 143, in get
    raise KeyError(f"Variable {key} does not exist")
KeyError: 'Variable PARQUET_DONE_PATH does not exist'
[2024-07-24T16:23:18.430+0900] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=make_parquet, task_id=to.parquet, run_id=scheduled__2024-07-12T00:00:00+00:00, execution_date=20240712T000000, start_date=20240724T072318, end_date=20240724T072318
[2024-07-24T16:23:18.433+0900] {standard_task_runner.py:110} ERROR - Failed to execute job 420 for task to.parquet ('Variable PARQUET_DONE_PATH does not exist'; 99063)
[2024-07-24T16:23:18.456+0900] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-07-24T16:23:18.463+0900] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-24T16:23:18.463+0900] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-07-24T16:34:48.992+0900] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-24T16:34:48.999+0900] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-24T16:34:49.001+0900] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-24T16:34:49.001+0900] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-24T16:34:49.005+0900] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): to.parquet> on 2024-07-12 00:00:00+00:00
[2024-07-24T16:34:49.007+0900] {standard_task_runner.py:64} INFO - Started process 543 to run task
[2024-07-24T16:34:49.009+0900] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'make_parquet', 'to.parquet', 'scheduled__2024-07-12T00:00:00+00:00', '--job-id', '375', '--raw', '--subdir', 'DAGS_FOLDER/make_parquet.py', '--cfg-path', '/var/folders/qv/0knwm5h50ml0skpppwpbks_r0000gp/T/tmpiwongkc2']
[2024-07-24T16:34:49.011+0900] {standard_task_runner.py:91} INFO - Job 375: Subtask to.parquet
[2024-07-24T16:34:49.024+0900] {task_command.py:426} INFO - Running <TaskInstance: make_parquet.to.parquet scheduled__2024-07-12T00:00:00+00:00 [running]> on host jeongui-MacBookPro.local
[2024-07-24T16:34:49.047+0900] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='make_parquet' AIRFLOW_CTX_TASK_ID='to.parquet' AIRFLOW_CTX_EXECUTION_DATE='2024-07-12T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-12T00:00:00+00:00'
[2024-07-24T16:34:49.047+0900] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-24T16:34:49.053+0900] {subprocess.py:63} INFO - Tmp dir root location: /var/folders/qv/0knwm5h50ml0skpppwpbks_r0000gp/T
[2024-07-24T16:34:49.053+0900] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n        echo "to.parquet"\n\n        READ_PATH=~/data/csv/20240712/csv.csv\n        SAVE_PATH=~/data/parquet/20240712/parquet.parquet\n\n        mkdir -p $SAVE_PATH\n\n        python ~/airflow/operator/csv2parquet.py $READ_PATH $SAVE_PATH\n        ']
[2024-07-24T16:34:49.056+0900] {subprocess.py:86} INFO - Output:
[2024-07-24T16:34:49.057+0900] {subprocess.py:93} INFO - to.parquet
[2024-07-24T16:34:49.509+0900] {subprocess.py:97} INFO - Command exited with return code 0
[2024-07-24T16:34:49.510+0900] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-24T16:34:49.517+0900] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=make_parquet, task_id=to.parquet, run_id=scheduled__2024-07-12T00:00:00+00:00, execution_date=20240712T000000, start_date=20240724T073448, end_date=20240724T073449
[2024-07-24T16:34:49.562+0900] {local_task_job_runner.py:243} INFO - Task exited with return code 0
[2024-07-24T16:34:49.571+0900] {taskinstance.py:3503} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-07-24T16:34:49.571+0900] {local_task_job_runner.py:222} INFO - ::endgroup::
